scipy/stats/_morestats.py

```diff
@@ -1861,19 +1861,19 @@ def shapiro(x):
       and false negatives (failure to reject a false null hypothesis).
 
     """
-    x = np.ravel(x)
+    x = np.ravel(x).astype(np.float64)
 
     N = len(x)
     if N < 3:
         raise ValueError("Data must be at least length 3.")
 
     x = x - np.median(x)
```

Comment:
We can delete this and subtract from `y` below to save memory and an unnecessary `median` calculation.

Reply:
Done.